src/spatialdata/_core/query/relational_query.py

```diff
@@ -129,6 +147,255 @@ def _filter_table_by_elements(
+            else:
+                warnings.warn(
+                    f"The element `{name}` is not annotated by the table. Skipping", UserWarning, stacklevel=2
+                )
+                element_dict[element_type][name] = None
+                continue
+    joined_table = table[joined_indices, :].copy() if joined_indices is not None else None
+    return element_dict, joined_table
+
+
+def _left_exclusive_join_spatialelement_table(
+    element_dict: dict[str, dict[str, Any]], table: AnnData
+) -> tuple[dict[str, Any], AnnData | None]:
+    regions, region_column_name, instance_key = get_table_keys(table)
+    groups_df = table.obs.groupby(by=region_column_name)
+    for element_type, name_element in element_dict.items():
+        for name, element in name_element.items():
+            if name in regions:
+                group_df = groups_df.get_group(name)
+                table_instance_key_column = group_df[instance_key]
+                if element_type in ["points", "shapes"]:
+                    mask = np.full(len(element), True, dtype=bool)
+                    mask[table_instance_key_column.values] = False
+                    masked_element = element.iloc[mask, :] if mask.sum() != 0 else None
+                    element_dict[element_type][name] = masked_element
+                else:
+                    warnings.warn(
+                        f"Element type `labels` not supported for left exclusive join. Skipping `{name}`",
+                        UserWarning,
+                        stacklevel=2,
+                    )
+                    continue
+            else:
+                warnings.warn(
+                    f"The element `{name}` is not annotated by the table. Skipping", UserWarning, stacklevel=2
+                )
+                continue
+
+    return element_dict, None
+
+
+def _left_join_spatialelement_table(
+    element_dict: dict[str, dict[str, Any]], table: AnnData
+) -> tuple[dict[str, Any], AnnData]:
+    regions, region_column_name, instance_key = get_table_keys(table)
+    groups_df = table.obs.groupby(by=region_column_name)
+    joined_indices = None
+    for element_type, name_element in element_dict.items():
+        for name, element in name_element.items():
+            if name in regions:
+                group_df = groups_df.get_group(name)
+                table_instance_key_column = group_df[instance_key]  # This is always a series
+                if element_type in ["points", "shapes"]:
+                    element_indices = element.index
+                else:
+                    element_indices = _get_unique_label_values_as_index(element)
+
+                mask = table_instance_key_column.isin(element_indices)
+                if joined_indices is None:
+                    joined_indices = table_instance_key_column[mask].index
+                else:
+                    # in place append does not work with pd.Index
```

Comment:
This doesn't preserve the order of the rows of the table but makes the table adopt the ordering given by the elements. This could be the behavior given when `match_rows_order=True`, but otherwise I think we should return the filtered table without reordering.

Reply:
it does, see [e7f36f2](https://github.com/scverse/spatialdata/pull/445/commits/e7f36f2fde1f34994c93a3861d95ebfe54960e5d)